From 1e999b44e67b3300bc7b693e82d9b41f7e803186 Mon Sep 17 00:00:00 2001
From: Hardik T Shah <hardik.t.shah@intel.com>
Date: Thu, 17 Mar 2016 11:19:29 +0530
Subject: [PATCH 0143/1082] REVERTME:SDW:CNL: Register only 3 master controller
 to bus.

There is constant BUS errors from the slaves on the 4th
SDW bus. This results in kernel dumps, as bus driver reports
the bus errors. This is a FPGA slave issue. So when FPGA
slave is connected register only 3 master controllers.

Change-Id: I0c61d5611ddf51dc56ce87e7b1e389d6638698bc
Signed-off-by: Hardik T Shah <hardik.t.shah@intel.com>
---
 sound/soc/intel/skylake/cnl-sst.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sound/soc/intel/skylake/cnl-sst.c b/sound/soc/intel/skylake/cnl-sst.c
index dc2030f5a2f2..e6e1023e86f0 100644
--- a/sound/soc/intel/skylake/cnl-sst.c
+++ b/sound/soc/intel/skylake/cnl-sst.c
@@ -509,7 +509,12 @@ static int skl_register_sdw_masters(struct device *dev, struct skl_sst *dsp,
 	struct cnl_sdw_data *p_data;
 	int ret = 0, i, j;
 	/* TODO: This number 4 should come from ACPI */
+#ifdef CONFIG_SDW_MAXIM_SLAVE
+
+	dsp->num_sdw_controllers = 3;
+#else
 	dsp->num_sdw_controllers = 4;
+#endif
 	master = devm_kzalloc(dev,
 			(sizeof(*master) * dsp->num_sdw_controllers),
 			GFP_KERNEL);
-- 
2.14.4

